src/main/java/edu/kit/hci/soli/config/security/SoliUserPrincipal.java

```diff
@@ -0,0 +1,55 @@
+package edu.kit.hci.soli.config.security;
+
+import edu.kit.hci.soli.domain.User;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.oauth2.core.oidc.OidcIdToken;
+import org.springframework.security.oauth2.core.oidc.OidcUserInfo;
+import org.springframework.security.oauth2.core.oidc.user.OidcUser;
+
+import java.util.List;
+import java.util.Map;
+
+public class SoliUserPrincipal implements OidcUser, SoliUserDetails {
```

Comment:
I guess this is an ok approach... When we have new "kinds" users we would simply add them here right?